x-pack/plugins/cross_cluster_replication/server/services/license.ts

```diff
@@ -5,83 +5,103 @@
  * 2.0.
  */
 
+import { i18n } from '@kbn/i18n';
 import { Logger } from 'src/core/server';
 import { KibanaRequest, KibanaResponseFactory, RequestHandler } from 'src/core/server';
+
+import type { LicenseType, LicenseCheckState } from '../../../licensing/common/types';
 import type { CcrRequestHandlerContext } from '../types';
 
 import { LicensingPluginSetup } from '../../../licensing/server';
-import { LicenseType } from '../../../licensing/common/types';
-
-export interface LicenseStatus {
-  isValid: boolean;
-  message?: string;
-}
+import { PLUGIN } from '../../common/constants';
 
 interface SetupSettings {
   pluginId: string;
   minimumLicenseType: LicenseType;
-  defaultErrorMessage: string;
 }
 
 export class License {
-  private licenseStatus: LicenseStatus = {
-    isValid: false,
-    message: 'Invalid License',
-  };
+  private licenseCheckState: LicenseCheckState = 'unavailable';
+  private licenseType?: LicenseType;
+  private logger?: Logger;
 
   private _isEsSecurityEnabled: boolean = false;
 
   setup(
-    { pluginId, minimumLicenseType, defaultErrorMessage }: SetupSettings,
+    { pluginId, minimumLicenseType }: SetupSettings,
     { licensing, logger }: { licensing: LicensingPluginSetup; logger: Logger }
   ) {
+    this.logger = logger;
+
     licensing.license$.subscribe((license) => {
```

Comment:
not directly related to this PR - but I noticed while looking at the code in my IDE that `LicensingPluginSetup.license$` is deprecated, and we should be using `license$` provided by the start contract.

Reply:
Good point! Done.